src/hooks/useWithdrawLiquidityHandler.tsx

```diff
@@ -22,342 +18,201 @@ export const useWithdrawLiquidityHandler = () => {
-        }
-        let tx = await posManager.withdrawReserves(WithdrawReservesParams)
-        return await tx.wait()
-      } catch (e) {
-        return e
+    try {
+      if (!posManager) {
+        await initializePositionManagerContract()
       }
-    } else {
-      if (depPool === null) {
-        return
+
+      const WithdrawReservesParams = {
+        cfmm: process.env.NEXT_PUBLIC_CFMM_ADDRESS,
+        protocol: 1,
+        amount: amt,
+        amountsMin: [100, 200, 300],
+        to: accountInfo.address,
+        deadline: ethers.constants.MaxUint256,
       }
 
-      try {
-        let tx = await depPool.removeLiquidity(amt, ZEROMIN, ZEROMIN, accountInfo.address, {
-          gasLimit: process.env.NEXT_PUBLIC_GAS_LIMIT,
-        })
+      if (posManager) {
+        let tx = await posManager.withdrawReserves(WithdrawReservesParams)
         return await tx.wait()
-      } catch (e) {
-        return e
       }
+    } catch (e) {
+      throw new Error('An error occurred while whithdrawing reserves')
     }
   }
 
-  async function approveWithdraw(posManager: Contract | null, contractAddress: string) {
+  async function approveWithdraw() {
+    if (!provider) {
+      notifyError('Please connect wallet')
+      return
+    }
+
     if (!accountInfo || !accountInfo.address) {
-      notifyError('Wallet not connected.')
+      notifyError('User wallet not found.')
       return
     }
-    if (process.env.NEXT_PUBLIC_ENVIRONMENT === 'local') {
-      if (posManager === null) {
-        return null
-      } else {
-        if (provider !== null) {
-          try {
-            let address = process.env.NEXT_PUBLIC_GAMMAPOOL_ADDRESS
-            let gammaPoolContract
-            if (provider && address) {
-              if (accountInfo && accountInfo?.address) {
-                gammaPoolContract = new ethers.Contract(
-                  address,
-                  TestGammaPool.abi,
-                  provider.getSigner(accountInfo?.address)
-                )
-              } else {
-                gammaPoolContract = new ethers.Contract(address, TestGammaPool.abi, provider)
-              }
-              let tx = await gammaPoolContract.approve(contractAddress, constants.MaxUint256.toString())
-              return await tx.wait()
-            } else {
-              notifyError('Please connect wallet')
-            }
-          } catch (e) {
-            throw e
-          }
-        } else {
-          notifyError('Please connect wallet')
-        }
-      }
-    } else {
-      if (depPool === null) {
-        return null
-      } else {
-        if (provider !== null) {
-          try {
-            let tx = await depPool.approve(contractAddress, constants.MaxUint256.toString())
-            return await tx.wait()
-          } catch (e) {
-            throw e
-          }
-        } else {
-          notifyError('Please connect wallet')
-        }
-      }
+
+    if (parseInt(totalLiquidityAmt) === 0) {
```

Comment:
surround with try catch and deal with exceptions

Reply:
Done